worlds/witness/player_items.py

```diff
@@ -149,43 +149,81 @@ def get_filler_items(self, quantity: int) -> Dict[str, int]:
         """
         Returns items that are ideal for placing on extremely early checks, like the tutorial gate.
         """
-        output: Set[str] = set()
-        if self._world.options.shuffle_symbols:
-            output = {"Dots", "Black/White Squares", "Symmetry", "Shapers", "Stars"}
+        output: Dict[str, List[str]] = {}
+
+        existing_items_lookup = {existing_item.name for existing_item in existing_items}
+
+        if self._world.options.shuffle_symbols and "Symbol" in self._world.options.early_good_items.value:
+            symbols = ["Dots", "Black/White Squares", "Symmetry", "Shapers", "Stars"]
 
             if self._world.options.shuffle_discarded_panels:
                 if self._world.options.puzzle_randomization == "sigma_expert":
-                    output.add("Arrows")
+                    symbols.append("Arrows")
                 else:
-                    output.add("Triangles")
+                    symbols.append("Triangles")
 
             # Replace progressive items with their parents.
-            output = {static_witness_logic.get_parent_progressive_item(item) for item in output}
-
-        # Remove items that are mentioned in any plando options. (Hopefully, in the future, plando will get resolved
-        #   before create_items so that we'll be able to check placed items instead of just removing all items mentioned
-        #   regardless of whether or not they actually wind up being manually placed.
-        for plando_setting in self._multiworld.plando_items[self._player_id]:
-            if plando_setting.get("from_pool", True):
-                for item_setting_key in [key for key in ["item", "items"] if key in plando_setting]:
-                    if isinstance(plando_setting[item_setting_key], str):
-                        output -= {plando_setting[item_setting_key]}
-                    elif isinstance(plando_setting[item_setting_key], dict):
-                        output -= {item for item, weight in plando_setting[item_setting_key].items() if weight}
-                    else:
-                        # Assume this is some other kind of iterable.
-                        for inner_item in plando_setting[item_setting_key]:
-                            if isinstance(inner_item, str):
-                                output -= {inner_item}
-                            elif isinstance(inner_item, dict):
-                                output -= {item for item, weight in inner_item.items() if weight}
-
-        # Sort the output for consistency across versions if the implementation changes but the logic does not.
-        return sorted(output)
+            symbols = [
+                static_witness_logic.get_parent_progressive_item(item) for item in symbols
+            ]
+
+            output["Symbol"] = symbols
+
+        if self._world.options.shuffle_doors and "Door / Door Panel" in self._world.options.early_good_items.value:
+            door_set = {
```

Comment:
What's the rationale behind what to include here? Some of these unlock one, or even zero locations

Reply:
Which ones unlock zero?

Reply:
Doors that _can_ unlock zero locations:
Caves Shortcuts, Desert Light Room Entry (Door), Caves Swamp/Mountain Shortcut (Door), Town Panels (because it doesn't exist), Glass Factory Entry (Panel), Windmill & Theater Control Panels.
And then quite a few often only unlock one location. Personally, I feel like these should be doors that _feel_ rewarding, so things like Swamp Laser Shortcut and Town Tower Doors don't really fit that.
I also feel like Caves Shortcuts might be best excluded since they have their own option, in a way. Or rather, if somebody _wanted_ them early, they could do that already.

Reply:
Problem is that in some of these modes (`disable_non_randomized` + `shuffle_symbols`), there aren't a lot of doors that actually do anything... that's mostly why these one-doors are here

Reply:
I changed it a bit further and I think now it's a lot better. Still happy to get more input though.